src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,169 @@
+    ----------
+    fom
+        The full-order |LTIModel| to reduce.
+    mu
+        |Parameter values|.
+    """
+
+    def __init__(self, fom, mu=None):
+        assert isinstance(fom, LTIModel)
+        if not isinstance(mu, Mu):
+            mu = fom.parameters.parse(mu)
+        assert fom.parameters.assert_compatible(mu)
+        self.fom = fom
+        self.mu = mu
+        self.V = None
+        self.W = None
+        self._pg_reductor = None
+
+    def reduce(self, r=None, decomposition='samdp', projection='orth',
+               symmetric=False, which="LR", method_options=None):
+        """Modal Truncation.
+
+        Parameters
+        ----------
+        r
+            Order of the reduced model.
+        decomposition
+            Algortihm to use for the decomposition:
+
+            - `'eig'`: use scipy.linalg.eig algorithm
+            - `'samdp'`: find dominant poles using samdp algorithm
+        projection
+            Projection method:
+
+            - `'orth'`: projection matrices are orthogonalized with
+              respect to the Euclidean inner product
+            - `'biorth'`: projection matrices are biorthogolized with
+              respect to the E product
+        symmetric
+            True if Operator A is symmetric and E is symmetric, positive
+            definite, False if not.
+        method_options
+            Optional dict with more options for the samdp algorithm.
+        which
+            A string specifying which `k` eigenvalues and eigenvectors to
+            compute when using the eig decomposition, default is "LR":
+
+            - `'SM'`: select eigenvalues with smallest magnitude
+            - `'LR'`: select eigenvalues with largest real part
+            - `'NR'`: select eigenvalues with largest norm(residual) / abs(Re(pole))
+            - `'NS'`: select eigenvalues with largest norm(residual) / abs(pole)
+            - `'NM'`: select eigenvalues with largest norm(residual)
+             
+        Returns
+        -------
+        rom
+            Reduced-order model.
+        """
+        assert 0 < r < self.fom.order
+        assert projection in ('orth', 'biorth')
+        assert decomposition in ('eig', 'samdp')
+        assert method_options is None or isinstance(method_options, dict)
+        if not method_options:
+            method_options = {}
+
+        self.V = self.fom.B.as_range_array().empty(reserve=r)
+        self.W = self.fom.C.as_source_array().empty(reserve=r)
+
+        if decomposition == 'eig':
+            if self.fom.A.sparse:
+                self.logger.warning('Converting operator A to a NumPy array.')
+            A = to_matrix(self.fom.A, format='dense')
+            if not isinstance(self.fom.E, IdentityOperator):
+                if self.fom.E.sparse:
+                    self.logger.warning('Converting operator E to a NumPy array.')
+            E = None if isinstance(self.fom.E, IdentityOperator) else to_matrix(self.fom.E, format='dense')
```

Comment:
To be consistent with `LTIModel.poles`, it would be good to use `pymor.models.iosys.sparse_min_size` in a similar way. You can do `from pymor.models.iosys import sparse_min_size` at the top and then just copy-paste the code from `LTIModel.poles`.